ipa.go

```diff
@@ -58,8 +58,11 @@ func toFrMultiple(res []*Fr, ps []*Point) {
 }
 
 func FromLEBytes(fr *Fr, data []byte) {
+	if len(data) > 32 {
+		panic("data is too long")
+	}
 	var aligned [32]byte
-	copy(aligned[:len(data)], data)
+	copy(aligned[:], data)
 	fr.SetBytesLE(aligned[:])
 }
```

Comment:
- I added the extra length check since that should hold true. If this doesn't hold, we have a bug somewhere in our logic. Truncating `data` in the `copy` doesn't sound like the right to do in any case.
- I removed that subslice truncation for two reasons: 1. Copy will only copy up to a minimum length of arguments anyway, 2. The `data` length check assumption must never result in `copy` copying less data than expected.

Reply:
Yes, we also have panics in other places which feels uncomfortable for the same reason. 
I think at some point returning an error would mean a complete halt in the geth node since it can't continue with the execution pipeline and is non-recoverable; but definitely a panic is blowing up `defers` and other nicier "good stops" for sure.

I'll create an issue to reconsider all panics in the package, since probably returning errors can have some breaking in geth that maybe we can batch at the same time.

Reply:
BTW, we should do the same in `go-ipa`; scan everything looking for panics and handle them differently.  Just to remember.